tools/privacyidea-cron

```diff
@@ -0,0 +1,135 @@
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import sys
+
+__doc__ = """
+This script is meant to be invoked periodically by the system cron daemon.
+It runs periodic tasks that are specified in the database.
+"""
+__version__ = "0.1"
+
+from datetime import datetime
+
+import dateutil
+from flask_script import Manager
+
+from privacyidea.app import create_app
+from privacyidea.lib.periodictask import get_scheduled_periodic_tasks, set_periodic_task_last_run, get_periodic_task_by_name, \
+    execute_task
+from privacyidea.lib.error import ParameterError
+
+app = create_app(config_name='production', silent=True)
+manager = Manager(app)
+
+
+def get_node_name(node):
+    """
+    Determine the node name. If no node name is given, read it from the app config.
+    Raise a ParameterError if no node name could be determined.
+    :param node: node name given by the user (can be None)
+    :return:
+    """
+    if node is None:
+        if "PI_AUDIT_SERVERNAME" in app.config:
+            return app.config["PI_AUDIT_SERVERNAME"]
+        else:
+            raise ParameterError("No PI_AUDIT_SERVERNAME specified in pi.cfg, use --node to pass a node name")
+    else:
+        return node
+
+
+def run_task_on_node(ptask, node):
+    """
+    Run a periodic task (given as a dictionary) on the given node.
+    In case of success, write the last successful run to the database. Catch any exceptions.
+    :param ptask: task as a dictionary
+    :param node: Node name
+    """
+    try:
+        print u"Running {!r} ...".format(ptask["name"]),
+        result = execute_task(ptask["taskmodule"], ptask["options"])
+    except Exception as e:
+        print u'Caught exception: {!r}'.format(e)
+        result = False
+    if result:
+        current_time = datetime.now(dateutil.tz.tzlocal())
+        print u'Task {!r} on node {!r} exited successfully. Noting this in the database ...'.format(ptask["name"], node)
+        set_periodic_task_last_run(ptask["id"], node, current_time)
+    else:
+        print u'Task {!r} on node {!r} did not run successfully.'.format(ptask["name"], node)
+        print u'This unsuccessful run is not recorded in the database.'
+    return result
+
+
+@manager.option("-n", "--node",
+                help="Override the node name (read from privacyIDEA config by default)",
+                dest="node_string")
+@manager.option("-t", "--task",
+                help="Run the specified task",
+                required=True,
+                dest="task_name")
+def run_manually(node_string, task_name):
```

Comment:
Nice idea to allow to run tasks manually.
In this case it would be good, to also have a command to list the configured tasks, so that the admin can list the tasks and choose which one to run.